templates/README.md

```diff
@@ -0,0 +1,276 @@
+
+Please note that the cnv.io suffix used in labels and annotations is temporary and is likely to change.
+
+## User Experience
+
+The flow should be the same as usual with regards to openshift templates. The user either takes a local template and generates a VM object.
+
+Or the user gets a template from kubernetes and invokes the proper REST command on top of it to keep the VM linked to the template or edit it in accordance to the template.
+
+The UI can then operate on top of template parameters and/or the Template.VM.spec DOM directly. There is an openshift [REST endpoint for instantiating Templates](https://docs.openshift.com/container-platform/3.7/rest_api/examples.html#template-instantiation) and the UI might be able to use it to create an in memory instance of VM it will then work with before posting it (kubernetes dry-run issues [https://github.com/kubernetes/kubernetes/issues/11488](#11488) and [https://github.com/kubernetes/kubernetes/issues/63559](#63559)).
+
+## Example file
+
+### template.yaml
+
+```YAML
+# vim: sts=2 sw=2 et
+apiVersion: v1
+kind: Template # Openshift kind of template or something "better"
+metadata:
+  name: windows-10
+  annotations:
+    openshift.io/display-name: "Generic Microsoft Windows 7+ template"
+    description: "Basic windows template long description with details"
+    openshift.io/long-description: >-
+      Long description of the template
+    openshift.io/provider-display-name: "Red Hat, Inc."
+    openshift.io/documentation-url: "https://cnv.io/..."
+    openshift.io/support-url: "https://access.redhat.com"
+    iconClass: icon-windows    
+
+    # The `defaults` set of annotations is meant as a hint only
+    # and is not going to be processed by the stock openshift templating
+    # mechanism. CNV UI can and should use it though.
+    # The information encoded there informs the tooling (UI)
+    # about names of devices that are to be used as templates
+    # for adding additional disk, networks, volumes and such
+
+    # The goal of default disk is to define what kind of disk
+    # is supported by the OS mainly in terms of bus (ide, scsi,
+    # sata, virtio, ...)
+    defaults.template.cnv.io/disk: default-disk
+
+    # The goal of default volume is to be able to configure mostly
+    # performance parameters like caches if those are exposed
+    # by the underlying volume implementation.
+    defaults.template.cnv.io/volume: default-volume
+
+    # The goal of default network is similar to default-disk
+    # and should be used as a template to ensure OS compatibility
+    # and performance
+    defaults.template.cnv.io/nic: default-nic
+
+    # The goal of default network is similar to default-volume
+    # and should be used as a template that specifies performance
+    # and connection parameters (L2 bridge for example)
+    defaults.template.cnv.io/network: default-network
```

Comment:
What parameteres are expected to be set here?

Reply:
Nothing for the stock templates (or maybe the L2 request), but the user can provide his own and there.. expected bandwith? network name?